woocommerce/woocommerce-ios
# [Shipping labels] Use backend data for saved packages

#14506


## Description


- Added WooShippingAddPackageViewModel, that model will be used for loading and using packages from the backend response
- In this PR we are using the saved packages and in https://github.com/woocommerce/woocommerce-ios/pull/14518 we will use the carrier packages

## Testing information


- Install and set up the Woo Shipping extension on your store.
- Build and run the app with the revampedShippingLabelCreation feature flag enabled.
- Create an order with the processing status and at least one physical product.
- In the order details, select "Create Shipping Label."
- Store options will start fetching from backend in the background.
- Tap "Select a Package" button
- Switch to "Saved" tab and check that are saved packages in the list, if there are none, be sure to add at least 1 custom package in the "custom" tab in the app or on web

## Screenshots


https://github.com/user-attachments/assets/76c7e624-bde3-4981-8a14-a23effb29203

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
